### PR TITLE
TS-2153 : Add -M to traffic_manager options if needed

### PR DESCRIPTION
**Pull request by strotyl**

### CODE REVIEW

**Comment by jpeach** [comment]
[approve ci]

File: mgmt/LocalManager.cc [hunk 1]
```diff
@@ -900,6 +900,10 @@ LocalManager::startProxy()
 
     real_proxy_options.append(proxy_options, strlen(proxy_options));
 
+    if (!strstr(proxy_options, "-M")) { // Make sure we're starting the proxy in mgmt mode
+      real_proxy_options.append(" -M ", strlen(" -M "));
```
Comment (jpeach) [inline mgmt/LocalManager.cc]: I don't think you need to append the trailing space.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/555/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/451/ for details.

File: mgmt/LocalManager.cc (hunk 1 above)
Comment (strotyl) [inline mgmt/LocalManager.cc]: I tested without the trailing space and it worked so I removed it.

**Comment by jpeach** [comment]
[approve ci]

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/557/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/453/ for details.

File: mgmt/LocalManager.cc [hunk 2]
```diff
@@ -900,6 +900,10 @@ LocalManager::startProxy()
 
     real_proxy_options.append(proxy_options, strlen(proxy_options));
 
+    if (!strstr(proxy_options, "-M")) { // Make sure we're starting the proxy in mgmt mode
```
Comment (SolidWallOfCode) [inline mgmt/LocalManager.cc]: This should really be a `#define` string to guarantee consistency and then use `sizeof(M_OPTION_STRING)-1` in the `append` call.

Comment (strotyl) [inline mgmt/LocalManager.cc]: Updated.

**Comment by jpeach** [comment]
[approve ci]

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/579/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/475/ for details.

**Comment by zwoop** [comment]
@jpeach we good to land this now?

**Pull request merged**